gorillaz.go

```diff
@@ -246,13 +248,23 @@ func (g *Gaz) Run() <-chan struct{} {
 		err = g.httpSrv.Serve(httpListener)
 		if err != nil {
 			if err != http.ErrServerClosed {
-				Sugar.Errorf("HTTP server server stopped on :%d, %v", httpPort, err)
-				panic(err)
-			} else {
-				Sugar.Infof("HTTP server server stopped on :%d", httpPort)
+				Log.Panic("HTTP serve stopped unexpectedly", zap.Error(err))
 			}
+			Sugar.Infof("HTTP server server stopped on :%d", httpPort)
 		}
 	}()
+	if g.ServiceDiscovery != nil {
+		Log.Info("registering service", zap.String("serviceName", g.ServiceName), zap.String("serviceAddr", g.serviceAddress), zap.Int("grpcPort", g.GrpcPort()), zap.Int("httpPort", g.HttpPort()))
+		var err error
+		g.registrationHandler, err = g.Register(&ServiceDefinition{ServiceName: g.ServiceName,
+			Addr:     g.serviceAddress,
+			GrpcPort: g.GrpcPort(),
+			HttpPort: g.HttpPort(),
```

Comment:
For @uportalis : what we have to keep in mind here is that in Kubernetes, we have to declare the Service exposing our gRPC and http endpoints on the same port as the pod, otherwise the service discovery in Consul will give us wrong ports (the one from the pod, which will not be accessible on the service).